mcstatus/pinger.py

```diff
@@ -3,53 +3,52 @@
 import datetime
 import json
 import random
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, TypedDict
+
+if TYPE_CHECKING:
+    from typing_extensions import NotRequired, TypeAlias
```

Comment:
Everywhere else in the code-base, this `if TYPE_CHECKING` block is below all other imports, not in the middle like this. Other than that though, the PR looks good to me and can be merged, needs resolving a conflict though.

Reply:
Weird, because external dependencies should come before mcstatus imports. Would isort complain if this was not in the conditional that it is in? That was my line of thinking in placing this here.

Reply:
Yeah, and indeed isort would complain with the import being below everything if it wasn't in an if block. However the reason I like defining these below other imports is that in many cases, these blocks also include variable definitions (like for example `P = ParamSpec("P")`, and in some cases it might even include entire class definitions (like with the typed-dicts in pinger.py). 

The alternative would be to use 2 blocks in these cases, one with the imports only, and the other one with these definitions, but I think having a single one below makes more sense, which is why I did it like that.

Reply:
It's back to where it was before. I completely removed the commits that modified mcstatus/pinger.py